blivet/fstab.py

```diff
@@ -0,0 +1,279 @@
+                                               exists=True),
+                                exists=True)
+        elif vfstype == "bind" or "bind" in mntops:
+            # bind mount... set vfstype so later comparison won't
+            # turn up false positives
+            vfstype = "bind"
+
+            # This is probably not going to do anything useful, so we'll
+            # make sure to try again from FSSet.mount_filesystems. The bind
+            # mount targets should be accessible by the time we try to do
+            # the bind mount from there.
+            parents = self._get_containing_device(spec, blivet.devicetree)
+            device = DirectoryDevice(spec, parents=parents, exists=True)
+            device.format = get_format("bind",
+                                       device=device.path,
+                                       exists=True)
+        elif file in ("/proc", "/sys", "/dev/shm", "/dev/pts",
+                      "/sys/fs/selinux", "/proc/bus/usb", "/sys/firmware/efi/efivars"):
+            # drop these now -- we'll recreate later
+            return None
+        else:
+            # nodev filesystem -- preserve or drop completely?
+            fmt = get_format(vfstype)
+            fmt_class = get_device_format_class("nodev")
+            if spec == "none" or \
+               (fmt_class and isinstance(fmt, fmt_class)):
+                device = NoDevice(fmt=fmt)
+
+        if device is None:
+            log.error("failed to resolve %s (%s) from fstab", spec,
+                      vfstype)
+            raise ValueError()
+
+        device.setup()
+        fmt = get_format(vfstype, device=device.path, exists=True)
+        if vfstype != "auto" and None in (device.format.type, fmt.type):
+            log.info("Unrecognized filesystem type for %s (%s)",
+                     device.name, vfstype)
+            device.teardown()
+            raise ValueError()
+
+        # make sure, if we're using a device from the tree, that
+        # the device's format we found matches what's in the fstab
+        ftype = getattr(fmt, "mount_type", fmt.type)
+        dtype = getattr(device.format, "mount_type", device.format.type)
+        if hasattr(fmt, "test_mount") and vfstype != "auto" and ftype != dtype:
+            log.info("fstab says %s at %s is %s", dtype, file, ftype)
+            if fmt.test_mount():     # pylint: disable=no-member
+                device.format = fmt
+            else:
+                device.teardown()
+                raise ValueError(_(
+                    "There is an entry in your /etc/fstab file that contains "
+                    "an invalid or incorrect file system type. The file says that "
+                    "{detected_type} at {mount_point} is {fstab_type}.").format(
+                    detected_type=dtype,
+                    mount_point=file,
+                    fstab_type=ftype
+                ))
+
+        del ftype
+        del dtype
```

Comment:
Is this necessary? I would expect Python to deal correctly with the memory management here. If not, a comment might be a good idea.

Reply:
Removed.

Reply:
Whole method was copied from anaconda. Removing `del`s does not seem to affect the behavior. Removed.